researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider2.java

```diff
@@ -0,0 +1,284 @@
+package org.sagebionetworks.bridge.researchstack;
+
+import android.content.Context;
+import android.support.annotation.AnyThread;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.researchstack.backbone.DataProvider;
+import org.researchstack.backbone.DataResponse;
+import org.researchstack.backbone.model.ConsentSignatureBody;
+import org.researchstack.backbone.model.SchedulesAndTasksModel;
+import org.researchstack.backbone.model.User;
+import org.researchstack.backbone.result.TaskResult;
+import org.researchstack.backbone.task.Task;
+import org.sagebionetworks.bridge.android.manager.BridgeManagerProvider;
+import org.sagebionetworks.bridge.android.manager.auth.AuthenticationManager;
+import org.sagebionetworks.bridge.rest.model.SharingScope;
+import org.sagebionetworks.bridge.rest.model.StudyParticipant;
+import org.sagebionetworks.bridge.rest.model.UserSessionInfo;
+
+import rx.Completable;
+import rx.Observable;
+import rx.Observer;
+import rx.functions.Action0;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sagebionetworks.bridge.researchstack.ApiUtils.SUCCESS_DATA_RESPONSE;
+
+/**
+ * Created by jyliu on 1/20/2017.
+ */
+@AnyThread
+public class BridgeDataProvider2 extends DataProvider {
```

Comment:
Unsure why I can't directly reply to your comment about Retrofit.

Retrofit comes from rest-client, which is Bridge's java SDK shared between Android and service code. It's generated via swagger, so theoretically we could switch away from Retrofit, but it would take some doing since a significant amount of Retrofit and OkHttp3 functionality has been built into rest-client.